### PR TITLE
FC-1057 fix for selectDistinct, offset, limit

### PR DESCRIPTION
**Pull request by ldw1007**
This is a "quick-fix" for a client-reported issue - more refactoring is needed; but, will be tracked under FC-1048.. comments welcome!

* changed process to 1) get-results, 2) apply sort (if orderBy), 3) apply limit/offset during sort when no groupBy, selectDistinct 4) format/expand results and then 5) if selectDistinct?, apply limit/offset
* changed commands used to selectDistinct to retain sort order
* fixed spelling in error message

### CODE REVIEW

File: src/fluree/db/query/fql.cljc [hunk 1]
```diff
@@ -618,16 +618,21 @@
...
-                   res     (<? (format-filter-tuples db tuples select-spec headers vars (dissoc opts :limit :offset :orderBy :groupBy)))]
+             (let [{:keys [headers tuples]} (if aggregates (analytical/add-aggregate-cols res aggregates) res)
+                   offset' (if (or selectDistinct? groupBy) 0 offset)
+                   limit'  (if selectDistinct? nil group-limit)
+                   tuples' (if orderBy
+                             (order-offset-and-limit-results tuples headers orderBy offset' limit')
+                             tuples)
+                   res'    (->> (dissoc opts :limit :offset :orderBy :groupBy)
+                                (format-filter-tuples db tuples' select-spec headers vars)
+                               <?)]
```
Comment (cap10morgan) [inline src/fluree/db/query/fql.cljc:629]: Indentation needs one more scoot to the right. :)

**Review by bplatz — APPROVED** [review]
Looks good!

File: src/fluree/db/query/fql.cljc [hunk 2]
```diff
@@ -618,16 +618,21 @@
...
                ;; TODO - drop unused columns, and calculate distinct before resolving all vals
                (if selectDistinct?
-                 (->> (into #{} res) (into []))
-                 res))))))
+                 (cond->> (distinct res')
+                          (and offset (< 0 offset)) (drop offset)
+                          (and group-limit (< 0 group-limit)) (take group-limit))
+                 res'))))))
```
Comment (cap10morgan) [inline src/fluree/db/query/fql.cljc:635]: If I'm reading this right, you could end up applying the offset and limit twice if you have both orderBy and selectDistinct? Is that a known issue that you're saving for later refactoring, or am I reading it wrong?

**Review by cap10morgan — APPROVED** [review]
One teensy tiny formatting thing and a question about the behavior. No blockers though.

File: src/fluree/db/query/fql.cljc (hunk 2 above)
Comment (ldw1007) [reply src/fluree/db/query/fql.cljc:635]: Above that section of code, where we are applying the sort to the results, we actually clear out the values:

                   offset' (if (or selectDistinct? groupBy) 0 offset)
                   limit'  (if selectDistinct? nil group-limit)
                   tuples' (if orderBy
                             (order-offset-and-limit-results tuples headers orderBy offset' limit')
                             tuples)

File: src/fluree/db/query/fql.cljc (hunk 1 above)
Comment (ldw1007) [reply src/fluree/db/query/fql.cljc:629]: thanks - noted & will make change as I start to refactor a bit

**Pull request merged**